src/test/objectstore/chain_xattr.cc

```diff
@@ -230,13 +230,32 @@ TEST(chain_xattr, listxattr) {
   const string x(LARGE_BLOCK_LEN, 'X');
   const int y = 1234;
 
+  int orig_size = chain_listxattr(file, NULL, 0);
+  char *orig_buffer = NULL;
+  string orig_str;
+  if (orig_size) {
+    orig_buffer = (char*)malloc(orig_size);
+    chain_flistxattr(fd, orig_buffer, orig_size);
+    orig_str = string(orig_buffer);
+    orig_size = orig_str.size();
+  }
```

Comment:
@yangdongsheng you can use the boost for splitting the concatenated xattrs, like:
```c++
  using strings = std::vector<std::string>;
  std::string s;
  int size = chain_listxattr(file, NULL, 0);
  s.resize(size);
  chain_flistxattr(fd, &s[0], s.size());
  strings attrs;
  boost::split(attrs, s, boost::is_from_range('\0', '\0'));
  std::sort(attrs.begin(), attrs.end());
```